README.md

```diff
@@ -17,19 +17,137 @@ SPDX-License-Identifier: MIT
+The Spezi ML Swift Package includes modules that are helpful to integrate ML-related functionality in your application.
 
-For more information, please refer to the [API documentation](https://swiftpackageindex.com/StanfordSpezi/SpeziML/documentation).
+# Spezi Open AI
+
+A module that allows you to interact with GPT-based large language models (LLMs) from OpenAI within your Spezi application.
+
+## Setup
 
+### 1. Add Spezi ML as a Dependency
 
-## Spezi ML Modules
+First, you will need to add the SpeziML Swift package to
+[your app in Xcode](https://developer.apple.com/documentation/xcode/adding-package-dependencies-to-your-app#) or
+[Swift package](https://developer.apple.com/documentation/xcode/creating-a-standalone-swift-package-with-xcode#Add-a-dependency-on-another-Swift-package). When adding the package, select the `SpeziOpenAI` target to add.
 
-The Spezi ML Swift Package contains the following modules that are helpful to integrate ML-related functionality in your application.
+### 2. Register the Open AI Component
 
-### SpeziML
+> [!IMPORTANT]
+> If your application is not yet configured to use Spezi, follow the [Spezi setup article](https://swiftpackageindex.com/stanfordspezi/spezi/documentation/spezi/initial-setup) to set up the core Spezi infrastructure.
 
-A placeholder Swift Package Target.
+You can configure the `OpenAIComponent` in the `SpeziAppDelegate` as follows.
+
+```swift
+import Spezi
+import SpeziOpenAI
+
+
+class ExampleDelegate: SpeziAppDelegate {
+    override var configuration: Configuration {
+        Configuration {
+            OpenAIComponent(apiToken: "API_KEY", openAIModel: .gpt4)
+        }
+    }
+}
+```
+
+In the example above, we have configured the `OpenAIComponent` to use the GPT-4 model with a default API key. Note that the choice of model and API key are persisted across application launches. The `apiToken` and `openAIModel` can also be accessed and changed at runtime. 
+
+The `SpeziOpenAI` package also provides an `OpenAIAPIKeyOnboardingStep` that can be used to allow the user to provide their API key during the onboarding process instead (see `Examples` below). If using the `OpenAIAPIKeyOnboardingStep`, the `apiToken` property can be omitted here.
+
+> [!NOTE]  
+> You can learn more about a [`Component` in the Spezi documentation](https://swiftpackageindex.com/stanfordspezi/spezi/documentation/spezi/component).
+
+## Examples
+
+### Creating a Chat Interface
+
+In this example, we will create a chat interface that allows the user to converse with the model. Responses from the model will be streamed.
+
+```swift
+import OpenAI
+import SpeziOpenAI
+import SwiftUI
+
+struct OpenAIChatView: View {
+    @EnvironmentObject private var openAIComponent: OpenAIComponent
+    @State private var chat: [Chat]
+    
+    var body: some View {
+        ChatView($chat)
+    }
+    .onChange(of: chat) { _ in
```

Comment:
This view modifier seems to modify the body, leading to a compiler error.

Reply:
Thanks for catching this. It has been fixed.